packages/react/src/hooks/useTrackToggle.ts

```diff
@@ -65,7 +65,7 @@ export function useTrackToggle<T extends ToggleSource>({
   const clickHandler: React.MouseEventHandler<HTMLButtonElement> = React.useCallback(
     (evt) => {
       userInteractionRef.current = true;
-      toggle().finally(() => (userInteractionRef.current = false));
```

Comment:
resetting this to `false` was happening before the `onChange` callback was fired in some instances. 
Removing the reset is fine, I think, because we reset it after every `onChange` call anyways.

If we want to make sure, we could reset it in `catch` instead of in `finally`